tracer/src/Datadog.Trace/OTelMetrics/MetricReaderHandler.cs

```diff
@@ -0,0 +1,340 @@
+                Log.Warning("Measurement recorded for instrument '{InstrumentName}' but no MetricState provided. Measurement will be dropped.", instrument.Name);
+            }
+        }
+
+        public static void OnMeasurementRecordedDouble(Instrument instrument, double value, ReadOnlySpan<KeyValuePair<string, object?>> tags, object? state)
+        {
+            if (state is MetricState metricState)
+            {
+                metricState.RecordMeasurementDouble(value, tags);
+            }
+            else
+            {
+                // Log error - this should not happen if we follow OTel SDK pattern
+                Log.Warning("Measurement recorded for instrument '{InstrumentName}' but no MetricState provided. Measurement will be dropped.", instrument.Name);
+            }
+        }
+
+        // Overloads for different numeric types
+        public static void OnMeasurementRecordedByte(Instrument instrument, byte value, ReadOnlySpan<KeyValuePair<string, object?>> tags, object? state)
+        {
+            OnMeasurementRecordedLong(instrument, value, tags, state);
+        }
+
+        public static void OnMeasurementRecordedShort(Instrument instrument, short value, ReadOnlySpan<KeyValuePair<string, object?>> tags, object? state)
+        {
+            OnMeasurementRecordedLong(instrument, value, tags, state);
+        }
+
+        public static void OnMeasurementRecordedInt(Instrument instrument, int value, ReadOnlySpan<KeyValuePair<string, object?>> tags, object? state)
+        {
+            OnMeasurementRecordedLong(instrument, value, tags, state);
+        }
+
+        public static void OnMeasurementRecordedFloat(Instrument instrument, float value, ReadOnlySpan<KeyValuePair<string, object?>> tags, object? state)
+        {
+            OnMeasurementRecordedDouble(instrument, value, tags, state);
+        }
+
+        internal static IReadOnlyList<MetricPoint> GetMetricsForExport(bool clearAfterGet = false)
+        {
+            var allMetricPoints = new List<MetricPoint>();
+
+            foreach (var kvp in CapturedMetrics)
+            {
+                var metricState = kvp.Value;
+                var metricPoints = metricState.GetMetricPoints();
+                allMetricPoints.AddRange(metricPoints);
+            }
+
+            if (clearAfterGet)
+            {
+                CapturedMetrics.Clear();
+            }
+
+            return allMetricPoints;
+        }
+
+        // For testing only
+        internal static IReadOnlyList<MetricPoint> GetCapturedMetricsForTesting()
+        {
+            var allMetricPoints = new List<MetricPoint>();
+
+            foreach (var kvp in CapturedMetrics)
+            {
+                var metricState = kvp.Value;
+                var metricPoints = metricState.GetMetricPoints();
+                allMetricPoints.AddRange(metricPoints);
+            }
+
+            return allMetricPoints;
+        }
```

Comment:
nit: AFACIT this is identical
```suggestion
        internal static IReadOnlyList<MetricPoint> GetCapturedMetricsForTesting()
            => GetMetricsForExport(clearAfterGet: false);

```